sdk/go/sdk.go

```diff
@@ -168,7 +166,7 @@ type SDK interface {
 	SendMessage(chanID, msg, token string) error
 
 	// ReadMessages read messages of specified channel.
-	ReadMessages(chanID, token string) ([]mainflux.Message, error)
+	ReadMessages(chanID, token string) (messagesPageRes, error)
```

Comment:
You cannot use a private type for an interface return value. Please replace this with some public type and use `messagesPageRes` only internally in SDK in order to deserialize HTTP response.